lib/commands/debug/index.js

```diff
@@ -0,0 +1,91 @@
+const path = require('path');
+const SmapiClient = require('@src/clients/smapi-client');
+const { AbstractCommand } = require('@src/commands/abstract-command');
+const optionModel = require('@src/commands/option-model');
+const AuthorizationController = require('@src/controllers/authorization-controller');
+const ResourcesConfig = require('@src/model/resources-config');
+const CONSTANTS = require('@src/utils/constants');
+const profileHelper = require('@src/utils/profile-helper');
+const jsonView = require('@src/view/json-view');
+const fs = require('fs');
+const stringUtils = require('@src/utils/string-utils');
+const helper = require('./helper');
+
+class DebugCommand extends AbstractCommand {
+    name() {
+        return 'debug';
+    }
+
+    description() {
+        return 'debug the skill';
+    }
+
+    requiredOptions() {
+        return [];
+    }
+
+    optionalOptions() {
+        return ['skill-id', 'profile', 'debug-port', 'wait-for-attach', 'skillCodeRegion'];
+    }
+
+    handle(cmd, cb) {
+        const debugPort = cmd.debugPort || 5000;
+        const profile = profileHelper.runtimeProfile(cmd.profile);
+        const skillCodeRegion = cmd.skillCodeRegion || 'default';
+        const smapiClient = new SmapiClient({
+            profile,
+            doDebug: cmd.debug
+        });
+        const authorizationController = new AuthorizationController({
+            auth_client_type: 'LWA',
+            doDebug: this.doDebug
+        });
+
+        authorizationController.tokenRefreshAndRead(profile, (tokenErr, token) => {
+            if (tokenErr) {
+                return cb(tokenErr);
+            }
+            new ResourcesConfig(path.join(process.cwd(), CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG));
```

Comment:
I moved it to within the try catch block for getting the skill id. Let me know if thats ok. Because the skill id would be obtained based on the profile, I dont think making a separate check for the profile on Resources is needed